src/pages/workspace/expensifyCard/WorkspaceEditCardLimitPage.tsx

```diff
@@ -0,0 +1,152 @@
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AccessOrNotFoundWrapper from '@pages/workspace/AccessOrNotFoundWrapper';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/EditExpensifyCardLimitForm';
+
+type ConfirmationWarningTranslationPaths = 'workspace.expensifyCard.smartLimitWarning' | 'workspace.expensifyCard.monthlyLimitWarning' | 'workspace.expensifyCard.fixedLimitWarning';
+
+// TODO: remove when Onyx data is available
+const mockedCard = {
+    accountID: 885646,
+    availableSpend: 1000,
+    nameValuePairs: {
+        cardTitle: 'Test 1',
+        isVirtual: true,
+        limit: 2000,
+        limitType: CONST.EXPENSIFY_CARD.LIMIT_TYPES.SMART,
+    },
+    lastFourPAN: '1234',
+};
+
+type WorkspaceEditCardLimitPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.EXPENSIFY_CARD_LIMIT>;
+
+function WorkspaceEditCardLimitPage({route}: WorkspaceEditCardLimitPageProps) {
+    const {policyID, cardID} = route.params;
+    const {translate} = useLocalize();
+    const {inputCallbackRef} = useAutoFocusInput();
+    const styles = useThemeStyles();
+    const [isConfirmModalVisible, setIsConfirmModalVisible] = useState(false);
+
+    const [cardsList] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${policyID}_${CONST.EXPENSIFY_CARD.BANK}`);
+    const card = cardsList?.[cardID] ?? mockedCard;
+
+    const getPromptTextKey = useMemo((): ConfirmationWarningTranslationPaths => {
+        switch (card.nameValuePairs?.limitType) {
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.SMART:
+                return 'workspace.expensifyCard.smartLimitWarning';
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.FIXED:
+                return 'workspace.expensifyCard.fixedLimitWarning';
+            case CONST.EXPENSIFY_CARD.LIMIT_TYPES.MONTHLY:
+                return 'workspace.expensifyCard.monthlyLimitWarning';
+            default:
+                return 'workspace.expensifyCard.fixedLimitWarning';
+        }
+    }, [card.nameValuePairs?.limitType]);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const updateCardLimit = (newLimit: string) => {
+        setIsConfirmModalVisible(false);
+
+        // TODO: add API call when it's supported https://github.com/Expensify/Expensify/issues/407831
+
+        Navigation.goBack();
+    };
+
+    const submit = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.EDIT_EXPENSIFY_CARD_LIMIT_FORM>) => {
+        const currentLimit = card.nameValuePairs?.limit ?? 0;
+        const currentRemainingLimit = currentLimit - card.availableSpend;
+        const newRemainingLimit = Number(values[INPUT_IDS.LIMIT]) * 100 - currentRemainingLimit;
```

Comment:
```suggestion
        const newRemainingLimit = Number(values[INPUT_IDS.LIMIT]) * 100 - card.availableSpend;
```

I think it should like that

Reply:
One more thing, Why do we `* 100`

Reply:
@DylanDylann I've followed the calculations suggested [there](https://callstack-hq.slack.com/archives/C036QM0SLJK/p1720715245848479?thread_ts=1720454092.016029&cid=C036QM0SLJK).  Why do you think it should be changed?
When I tested it seems to work as expected...

About `* 100` - all money values are stored in cents I guess, so I turn the input value into cents to make the calculations correct. What do you think?

Reply:
> I've followed the calculations suggested [there](https://callstack-hq.slack.com/archives/C036QM0SLJK/p1720715245848479?thread_ts=1720454092.016029&cid=C036QM0SLJK). Why do you think it should be changed?
When I tested it seems to work as expected...

I can't access this reference. @mountiny Could you help to confirm this calculation?

Reply:
> About * 100 - all money values are stored in cents I guess, so I turn the input value into cents to make the calculations correct. What do you think?


@VickyStash  You mean that if users type $20, the `inputValue` will be `0.2`. I don't think this makes sense. But let's wait @mountiny take a look at this?

Reply:
> @VickyStash You mean that if users type $20, the inputValue will be 0.2. I don't think this makes sense. But let's wait @mountiny take a look at this?

I mean that if the user types $20 in the DB it will be stored as 2000.
And other values are also in cents, so that's why I do `* 100` during calculations.

But yeah, let's wait for @mountiny to be sure.

Reply:
@VickyStash Yeah. Make sense. Thanks

Reply:
These calculations can get a bit confusing (for me 😄) so let's use these terminologies:

- `availableSpend` - the amount that is available to spend on the card (currentLimit - currentSpend)
- `currentSpend` - the amount that the user has already spent on the card (currentLimit - availableSpend)
- `currentLimit` - the existing limit on the card (availableSpend + currentSpend)
- `newLimit` - the new limit that the admin is setting on the card
- `newAvailableSpend` - after updating the limit, the new spend that is available on the card (newLimit - currentSpend)

```js
const currentLimit = card.nameValuePairs?.limit ?? 0;
const currentSpend = currentLimit - card.availableSpend;
const newLimit = Number(values[INPUT_IDS.LIMIT]) * 100;
const newAvailableSpend = newLimit - currentSpend;
```

Let me know what all you think!

Reply:
Thank you @MariaHCD 🙌 !
I've updated it cc @DylanDylann